File: ez/merge_sorted.py

class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        target_loc = -1
        n1_p = m - 1
        n2_p = n - 1
        
        while n1_p >= 0 and n2_p >= 0:
            if nums1[n1_p] > nums2[n2_p]:
                nums1[target_loc] = nums1[n1_p]
                n1_p -= 1
            else:
                nums1[target_loc] = nums2[n2_p]
                n2_p -= 1
            target_loc -= 1
        
        #deal with residual
        while n2_p >= 0:
            nums1[target_loc] = nums2[n2_p]
            n2_p -= 1
            target_loc -= 1
            
        

File: ez/merge_sorted.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int p1 = m - 1, p2 = n - 1, p = m + n - 1;
        for(; p>= 0;p--)
        {
            if(p2 < 0)
                break;
            else{
                if(p1 >= 0 and nums1[p1] > nums2[p2])
                    nums1[p] = nums1[p1--];
                else
                    nums1[p] = nums2[p2--];
            }
        }
    }
};